release-pipeline.yml

```diff
@@ -0,0 +1,1994 @@
+  #             CPK_ENCRYPTION_KEY: $(CPK_ENCRYPTION_KEY)
+  #             CPK_ENCRYPTION_KEY_SHA256: $(CPK_ENCRYPTION_KEY_SHA256)
+  #             AZCOPY_E2E_EXECUTABLE_PATH: $(System.DefaultWorkingDirectory)/$(build_name)
+  #             GOCOVERDIR: '$(System.DefaultWorkingDirectory)/coverage'
+  #             NEW_E2E_SUBSCRIPTION_ID: $(AZCOPY_NEW_E2E_SUBSCRIPTION_ID)
+  #             NEW_E2E_APPLICATION_ID: $(AZCOPY_NEW_E2E_APPLICATION_ID)
+  #             NEW_E2E_CLIENT_SECRET: $(AZCOPY_NEW_E2E_CLIENT_SECRET)
+  #             NEW_E2E_TENANT_ID: $(OAUTH_TENANT_ID)
+  #             NEW_E2E_AZCOPY_PATH: $(System.DefaultWorkingDirectory)/$(build_name)
+  #           displayName: 'E2E Test $(display_name) - AMD64'
+
+  #         - task: PublishBuildArtifacts@1
+  #           displayName: 'Publish logs'
+  #           condition: succeededOrFailed()
+  #           inputs:
+  #             pathToPublish: '$(System.DefaultWorkingDirectory)/logs'
+  #             artifactName: logs
+
+  #         - task: PublishTestResults@2
+  #           condition: succeededOrFailed()
+  #           inputs:
+  #             testRunner: JUnit
+  #             testResultsFiles: $(System.DefaultWorkingDirectory)/**/$(display_name)_report.xml
+  #             testRunTitle: 'Go on $(display_name)'
+
+  #         - task: PublishCodeCoverageResults@1
+  #           condition: succeededOrFailed()
+  #           inputs:
+  #             codeCoverageTool: Cobertura
+  #             summaryFileLocation: $(System.DefaultWorkingDirectory)/**/$(display_name)_coverage.xml
+  #             additionalCodeCoverageFiles: $(System.DefaultWorkingDirectory)/**/$(display_name)_coverage.html
+
+  - stage: BuildArtifacts
+    #dependsOn: RunTests
+    #condition: succeeded('RunTests')
+    jobs:
+      - job: Set_1
+        timeoutInMinutes: 120
+        strategy:
+          matrix:
+            Ubuntu_amd64:
+              imageName: 'ubuntu-20.04'
+              container: 'test-cnt-ubn-20'
+              AgentName: "blobfuse-ubuntu20"
+        pool:
+          name: "blobfuse-ubuntu-pool"
+          demands:
+            - ImageOverride -equals $(agentName)
+
+        variables:
+          - name: root_dir
+            value: '$(System.DefaultWorkingDirectory)'
+          - name: work_dir
+            value: '$(System.DefaultWorkingDirectory)/azure-storage-azcopy'
+          - name: archives
+            value: $(root_dir)/archives
+          - name: linux_amd64
+            value: $(work_dir)/linux_amd64
+          - name: linux_se_amd64
+            value: $(work_dir)/linux_se_amd64
+
+        steps:
+          - script: |
+              wget https://raw.githubusercontent.com/Azure/azure-storage-azcopy/az-pipelineTest/go_installer.sh -P $(work_dir)/
+              chmod 777 $(work_dir)/go_installer.sh
+              sudo ls -lRt $(work_dir)/
+            displayName: 'Download go installer script'
+
+          # Custom script to install Go-lang
+          - task: ShellScript@2
```

Comment:
Why not install Go the default Azure Pipeline way?

Reply:
The reason we are using a custom script for go installation is to install go based on the system's architecture.

Reply:
Default step does not give you the latest version, on many linux distros. This is safer if we want to compile on any other distro than Ubn.